lib/auth/auth_with_roles.go

```diff
@@ -2417,17 +2417,17 @@ func (a *ServerWithRoles) CreateOIDCAuthRequest(req services.OIDCAuthRequest) (*
 		}
 	}
 
-	oidcReq, err := a.authServer.CreateOIDCAuthRequest(req)
+	oidcReq, err := a.authServer.CreateOIDCAuthRequest(ctx, req)
 	if err != nil {
-		emitSSOLoginFailureEvent(a.authServer.closeCtx, a.authServer.emitter, events.LoginMethodOIDC, err, req.SSOTestFlow)
+		emitSSOLoginFailureEvent(ctx, a.authServer.emitter, events.LoginMethodOIDC, err, req.SSOTestFlow)
```

Comment:
Hmm, I'm not too sure about using the request context here. Realistically, it's a very small window of opportunity, but it would mean that if a request timed out after an authentication failure, the authentication failure wouldn't go into the audit log.